src/components/Wiki/EntryEditForm/EntryEditForm.tsx

```diff
@@ -0,0 +1,236 @@
+        e.preventDefault();
+        updateEntry(this.state.entryData);
+    }
+
+    handleNewHeadingSubmit(e: React.FormEvent<HTMLFormElement>){
+        e.preventDefault();
+        postNewHeading(
+            this.state.newHeading,
+            {
+                context: this.state.newHeading.title,
+                textAdded: this.state.newHeading.text,
+                user: 1
+            },
+            this.props.entryData);
+    }
+
+    handleEntryChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        const entryInfo = this.state.entryData;
+        const newEntryInfo = {
+            ...entryInfo,
+            [e.target.name]: e.target.value
+        }
+        this.setState({entryData: newEntryInfo})
+    }
+
+    handleNewHeadingChange = (e: React.ChangeEvent<HTMLInputElement>) : void => {
+        const newHeadingInfo = this.state.newHeading;
+        const newNewHeadingInfo = {
+            ...newHeadingInfo,
+            [e.target.name]: e.target.value
+        }
+        this.setState({newHeading: newNewHeadingInfo})
+    }
+
+    handleSideBarChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        let sideBarContentString = JSON.stringify(this.state.sideBarData.content);
+
+        //replace value/parse to json
+        let adjusted = sideBarContentString.replace("\""+e.target.id+"\"", "\""+e.target.value+"\"");
+        let adjustedJSON = JSON.parse(adjusted);
+
+        this.setState({
+            sideBarData: {
+                ...this.state.sideBarData,
+                content: adjustedJSON
+            }
+        })
+    }
+
+    buildSideBarElements(){
+        if(this.state.sideBarData.content !== null) {
+            for (const [key, value] of Object.entries(this.state.sideBarData.content)) {
+                this.setState({
+                    sideBarElements: this.state.sideBarElements.concat(
+                        <div>
+                            <Form.Control className="float-left" id={key} value={key} onChange={this.handleSideBarChange}></Form.Control>
+                            <Form.Control className="float-right" id={value} value={value} onChange={this.handleSideBarChange}></Form.Control>
+                        </div>
+                    )
+                })
+            }
+        }
+    }
+
+    componentDidUpdate(prevProps: Readonly<entryEditFormProps>, prevState: Readonly<entryEditFormState>, snapshot?: any) {
+        if(prevProps.entryData.id !== this.props.entryData.id){
+            setTimeout( () => {
+                this.setState({
+                    entryData: this.props.entryData,
+                    sideBarData: this.props.sideBarData,
+                    sideBarElements: [],
+                    headingEditElements: this.props.headingEditElements
+                })
+                this.buildSideBarElements();
+            }, 200)
+        }
+        // else if(prevState.sideBarData.content !== this.state.sideBarData.content){
```

Comment:
Remove commented-out code if it is not going to be used.

Reply:
Going to use that for updating headings, so it will be used :thumbsup: